### ISSUE TITLE
Support for `extensions` field in kubeconfig

### ISSUE THREAD

**Issue by daniv-msft** [kubernetes-client/csharp#575]
Hello,

My team is working on [Bridge to Kubernetes](https://github.com/Microsoft/mindaro). We recently received multiple reports from users encountering issues because their kubeconfig contains an `extensions` field, and that the C# KubernetesClient rejects it: `Expected 'MappingStart', got 'SequenceStart' (at Line: 6, Col: 5, Idx: 124).`
Here is an example of such kubeconfig (from https://github.com/microsoft/mindaro/issues/111):
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority: C:\Users\REMOVED\.minikube\ca.crt
    extensions:
    - extension:
        last-update: Tue, 02 Feb 2021 13:45:29 GMT
        provider: minikube.sigs.k8s.io
        version: v1.17.0
      name: cluster_info
    server: https://127.0.0.1:49163
  name: minikube
contexts:
- context:
    cluster: minikube
    extensions:
    - extension:
        last-update: Tue, 02 Feb 2021 13:45:29 GMT
        provider: minikube.sigs.k8s.io
        version: v1.17.0
      name: context_info
    namespace: default
    user: minikube
  name: minikube
current-context: minikube
kind: Config
preferences: {}
users:
- name: minikube
  user:
    client-certificate: C:\Users\REMOVED\.minikube\profiles\minikube\client.crt
    client-key: C:\Users\REMOVED\.minikube\profiles\minikube\client.key
```

Could you please let me know if this is something you're aware of?
Would it be possible to add support for this field?

This isn't something we ever heard of until recently, and we're now seeing multiple occurrences of it. From what we've seen, this type of kubeconfig might be related to Minikube.

**Comment by brendandburns** [comment]
I think that this is fixed by:
https://github.com/kubernetes-client/csharp/pull/574

which will allow yamls with unknown tags to be parsed (and unknown tags ignored)

**Comment by brendandburns** [comment]
And if you want a PR to add this field would be welcome.

**Comment by daniv-msft** [comment]
Thank you @brendandburns for your quick reply! We'll give a try to the change you mention.
We've plenty to do on our side right now, but we'll definitely consider adding support for this tag as well if we have some bandwidth.

**Comment by tonyabbott** [comment]
It looks like this is the same issue fixed by #556
